akka-cluster/src/main/scala/akka/cluster/JoinConfigCompatChecker.scala

```diff
@@ -48,16 +48,42 @@ object JoinConfigCompatChecker {
       }
 
     if (result.isEmpty) Valid
-    else Invalid(result.to[im.Seq])
+    else Invalid(result)
+  }
+
+  /**
+   * Computes the difference of `actualConfig` and `toCheck` to determine
+   * whether new required keys are in `toCheck` and not in `actualConfig`.
+   *
+   * @param requiredKeys a Seq of required keys
+   * @param toCheck      the Config instance to be checked
+   * @param actualConfig the Config instance containing the expected values
+   */
+  def diff(requiredKeys: im.Seq[String], toCheck: Config, actualConfig: Config): ConfigValidation = {
+
+    def keys(c: Config) = c.entrySet().asScala.map(_.getKey).toSet
+    // simplify
+    val joiningKeys = keys(toCheck)
```

Comment:
it's slightly misleading to call this joiningKeys because the checks are done on both sides
1. joinConfigCompatChecker.check(joiningNodeConfig
1. joinConfigCompatChecker.check(clusterConfig

The latter is on the joining side but the parameter `toCheck` is the config of the existing cluster.

Reply:
Indeed. I saw those checks too. Updating.